packages/test/test-end-to-end-tests/src/test/SummarizeFetchValidation.spec.ts

```diff
@@ -0,0 +1,314 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import {
+    ContainerRuntime,
+    IContainerRuntimeOptions,
+    ISummarizer,
+    ISummarizeResults,
+} from "@fluidframework/container-runtime";
+import { IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider,
+    waitForContainerConnection,
+    summarizeNow,
+    createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, getContainerRuntimeApi } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { MockLogger } from "@fluidframework/telemetry-utils";
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import { FetchSource, ISummaryContext } from "@fluidframework/driver-definitions";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+import { pkgVersion } from "../packageVersion";
+
+// Note GC needs to be disabled.
+const runtimeOptions: IContainerRuntimeOptions = {
+    summaryOptions: {
+        disableSummaries: true,
+        summaryConfigOverrides: { state: "disabled" },
+    },
+    gcOptions: { gcAllowed: false },
+};
+export const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+
+class TestDataObject1 extends DataObject {
```

Comment:
I would suggest using the default data object provided by the TestObjectProvider. It has a root map that you can use to send ops. That will make the test simpler and also you don't have to worry about back-compat as it will be tested automatically. Here, the data store runtime will always be of the current version (N) and you lose compat testing.

Reply:
AS I plan to add  additional tests, it will give me more flexibility to do that.